packages/headless/src/controllers/search-parameter-manager/headless-search-parameter-manager.ts

```diff
@@ -108,19 +107,17 @@ function getAq(state: Partial<SearchParametersState>) {
     return {};
   }
 
-  const aq = state.advancedSearchQueries.aq;
-  const shouldInclude = aq !== getAdvancedSearchQueriesInitialState().aq;
-  return shouldInclude ? {aq} : {};
+  const {aq, aqWasSet} = state.advancedSearchQueries;
+  return aqWasSet ? {aq} : {};
 }
 
 function getCq(state: Partial<SearchParametersState>) {
   if (state.advancedSearchQueries === undefined) {
     return {};
   }
 
-  const cq = state.advancedSearchQueries.cq;
-  const shouldInclude = cq !== getAdvancedSearchQueriesInitialState().cq;
-  return shouldInclude ? {cq} : {};
+  const {cq, cqWasSet} = state.advancedSearchQueries;
+  return cqWasSet ? {cq} : {};
```

Comment:
The replacement of `shouldInclude` with `cqWasSet` means that manually setting the `cq`, even manually setting it to the default value, will include it in the search parameters.

I have a slight preference for having this behaviour, but I don't mind keeping the old behaviour if you prefer.

Reply:
I just got an idea looking at the new condition. Since it always adds the `cq` as long as it was set, it is no longer hiding the `cq` parameter when the value is equal to the default.

Isn't this in the same as having `shouldInclude` always set to `true`? If so, then it seems it would be simpler to just always include the `cq` value, and remove the conditional all together. This is in effect what the new code is doing?

Reply:
all urls using headless will contain `cq=` which can be pretty cryptic.

Reply:
Making the change only include `cq` at all times would cause the `initialState` property of the `Tab` controller to become useless. Thus, loading something like `https://somewebsite/search` would always select the "all" tab.

Reply:
Ideally we would only include the `cq` in the url if it's value is different from the initial cq value.

With the current changes is that the case?
Won't `cqWasSet` be `true` the moment a tab is [updated](https://github.com/coveo/ui-kit/pull/522/files#diff-afab97fef7d9ccf209a66b2972b76192ef6a00c27118f06f34710f42b6e36dccR23)?

If the initial tab is `All`, I click on `youtube`, then go back to `All`. I believe the cq will be in the url whereas before it would not have been?

Reply:
Yes, the last scenario you described is the case with this PR, this is what I tried to describe in my [first comment](https://github.com/coveo/ui-kit/pull/522#discussion_r582230466).

It may be possible to remove the `default` state value and keep this behaviour. However, if we want the `cq` to be able to disappear from the URL when it's equal to the default value, we need to keep it.

Reply:
If we want to stay consistent with the other url parameters, we would only add the `cq` to the url if the current `cq` expression is different from the initial value.

Reply:
Then I'll add back the default and compare against it